### ISSUE TITLE
Creating a Server Config Backup, deleting channel, restoring backup results in negative revision delta

### ISSUE THREAD

**Issue by rbeckman-nextgen** [nextgenhealthcare/connect#2654]
1) Create a channel
2) Create a Server Config Backup
3) Delete the channel
4) Restore the Server Config Backup

After the backup is restored and channels redeployed, the Rev Delta column on the Dashboard and Channel views shows a negative number.  The actual value of the number depends upon how many revisions had been made to the channel before deleting it.

Imported Issue. Original Details:
Jira Issue Key: MIRTH-2724
Reporter: ericb
Created: 2013-07-03T15:49:19.000-0700

**Comment by rbeckman-nextgen** [comment]
I followed these steps and was unable to reproduce this, in either 2.2.2 or RC 1...

Imported Comment. Original Details:
Author: narupley
Created: 2013-07-10T11:07:30.000-0700

**Comment by rbeckman-nextgen** [comment]
Eric, are you sure you got this behavior in 2.2.2 and not 2.2.1?  There was a change in 2.2.2 that may have fixed this issue.

Imported Comment. Original Details:
Author: jacobb
Created: 2013-07-10T11:17:03.000-0700

**Comment by rbeckman-nextgen** [comment]
I tried it on 2.2.1 and couldn't reproduce it there either. I also tried various permutations of saving / deploying channels within those steps as well.

Imported Comment. Original Details:
Author: narupley
Created: 2013-07-10T11:22:52.000-0700

**Comment by rbeckman-nextgen** [comment]
You know, I could have **sworn** I reproduced this on my local 2.2.2 instance, but now I can't.  However, I'm seeing it (and continue to see it) running MC 2.2.2 on the Appliances.  This is where I first discovered the problem, when walking through the backup/restore process on the Appliances in the training class.

If you'd like, you can try this out on one of the training appliances:

s16a.mirthcorp.com
admin
shoff-it-gak


Imported Comment. Original Details:
Author: ericb
Created: 2013-07-10T12:11:01.000-0700

**Comment by rbeckman-nextgen** [comment]
I just tried it on that appliance, and I can't reproduce it there either. Can you double-check the steps you laid out in the description and make sure that's _exactly_ what you did to cause the issue?

Imported Comment. Original Details:
Author: narupley
Created: 2013-07-10T12:38:15.000-0700

**Comment by rbeckman-nextgen** [comment]
OK, just tried it again on one of the training appliances, and I did skip an important step.

Between steps 1 and 2 listed in the description, DEPLOY the channel.

If you don't deploy it, the rev delta will be 0 (as expected) when the config is deployed.

I can show you on my computer if you are still unable to recreate.

Imported Comment. Original Details:
Author: ericb
Created: 2013-07-10T13:43:21.000-0700

**Comment by rbeckman-nextgen** [comment]
The issue definitely happens in 2.2.2, but should already be fixed in RC 1, and in the latest branches/2.x and trunk builds. I haven't been able to reproduce it anywhere but 2.2.2 and earlier.

Imported Comment. Original Details:
Author: narupley
Created: 2013-07-12T14:23:40.000-0700

**Comment by rbeckman-nextgen** [comment]
Eric, what was the appliance version we were using when we saw it happening?  The revision number would be useful.

Imported Comment. Original Details:
Author: jacobb
Created: 2013-07-12T22:07:54.000-0700

**Comment by rbeckman-nextgen** [comment]
3.6.6 - the latest

Imported Comment. Original Details:
Author: ericb
Created: 2013-07-12T22:13:26.000-0700

**Comment by rbeckman-nextgen** [comment]
Now when testing on the appliance, I assume you were using 2.2.2, not the latest 2.x build or any 3.0 version, right? If so that makes sense then, because I also was able to reproduce the issue in 2.2.2.

Imported Comment. Original Details:
Author: narupley
Created: 2013-07-12T23:29:18.000-0700

**Comment by rbeckman-nextgen** [comment]
Correct. 

Imported Comment. Original Details:
Author: ericb
Created: 2013-07-12T23:34:42.000-0700

**Comment by rbeckman-nextgen** [comment]
Nick, what about the issue of restoring a server config and seeing some revision deltas > 0 in the latest trunk?  Is that issue related or separate?  It sounds like this can be closed as resolved as a result of another issue.  What issue fixed it?

Imported Comment. Original Details:
Author: jacobb
Created: 2013-07-13T12:15:11.000-0700

**Comment by rbeckman-nextgen** [comment]
After reproducing this and testing it in 2.2.2 and the latest 2.x/3.0 builds, I've found that this particular issue has been resolved with the fixes made to the channel cache in both branches/2.x and trunk since 2.2.2 (MIRTH-2642). I'm not closing the issue only because I'd like someone else to double-check and verify.

There is actually a separate issue (MIRTH-2756) where restoring a server config can result in _positive_ revision deltas, but the root cause is different than this issue.

Imported Comment. Original Details:
Author: narupley
Created: 2013-07-13T20:23:24.000-0700
